batch/batch/driver/billing_manager.py

```diff
@@ -48,6 +54,69 @@ class CloudBillingManager(abc.ABC):
     product_versions: ProductVersions
     resource_rates: Dict[str, float]
 
+    async def _refresh_resources_from_retail_prices(self, prices: List[Price]):
+        log.info('refreshing resources from retail prices')
+        resource_updates = []
+        product_version_updates = []
+
+        for price in prices:
+            product = price.product
+            latest_product_version = price.version
+            latest_resource_rate = price.rate
+
+            resource_name = product_version_to_resource(product, latest_product_version)
+
+            current_product_version = self.product_versions.latest_version(product)
+            current_resource_rate = self.resource_rates.get(resource_name)
+
+            if CLOUD == 'gcp' and current_product_version and current_product_version != latest_product_version:
```

Comment:
Why is this specific to GCP? It's unclear to me why we need to do a cloud-specific check here and would love to remove it if possible.

Reply:
See comment above.